packages/as-types/assembly/safeMath.ts

```diff
@@ -1,8 +1,29 @@
+/**
+ * This class provides utility functions for basic arithmetic operations.
+ * These functions perform overflow and underflow checks to prevent unwanted behavior
+ * when dealing with unsigned 64-bit integers (u64). The SafeMath class should be used
+ * when working with arithmetic operations that require increased safety and precision.
+ *
+ * @remarks
+ * The SafeMath class is designed to be a drop-in replacement for standard arithmetic
+ * operations on unsigned 64-bit integers. By using the methods provided by this class,
+ * developers can avoid potential overflow and underflow issues and ensure that their
+ * code behaves correctly even in edge cases.
+ */
 export class SafeMath {
   /**
+   * Safely adds two unsigned 64-bit integers (u64), reverting if an
+   * overflow occurs.
    *
-   * @returns Returns the addition of two unsigned integers,
-   * reverting on overflow.
+   * @remarks
+   * This function is used to safely add two unsigned 64-bit integers without causing an overflow.
+   *
+   * @param a - The first operand for addition.
+   * @param b - The second operand for addition.
+   *
+   * @returns The sum of a and b as an unsigned 64-bit integer (u64).
+   *
+   * @throws Throws an error if the operation results in an overflow.
```

Comment:
To be consistent with what [we did for as-sdk](https://github.com/massalabs/massa-as-sdk/blob/main/assembly/std/storage.ts#L170-L173):
```suggestion
   * @throws if the operation results in a number bigger than u64.MAX_VALUE.
```